core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -984,12 +984,12 @@ object IO {
     v.fold[IO[Unit, A]](IO.fail(()))(IO.now)
 
   /**
-   * Lifts a `Try` into an `IO`.
+   * Imports a `Try` into an `IO`.
    */
-  final def fromTry[A](v: scala.util.Try[A]): IO[Throwable, A] =
-    v match {
+  final def fromTry[A](effect: => scala.util.Try[A]): IO[Throwable, A] =
+    sync(effect).flatMap {
```

Comment:
Change this to `syncThrowable` and it's good to go! That way if creating the `Try` throws an exception, we can report that in the returned `IO`.